test/environments/rails50/Gemfile

```diff
@@ -8,7 +8,7 @@ gem 'rails', '~> 5.0.0'
 gem 'minitest', '5.2.3'
 gem 'minitest-stub-const', '~> 0.6'
 gem 'mocha', '~> 1.1.0', :require => false
-gem 'rack', '< 2.1.0'
+gem 'rack', "~> #{RUBY_VERSION >= '2.3.0' ? '2.2.4' : '2.1.4.1'}"
```

Comment:
This is only tested on JRuby at this time, so the condition may not be necessary

Reply:
Ah okay, cool. I was just drawing a line between Rails 7/edge and all others and giving all the others this same treatment.